File: tutorials/b1-checkpoint-restart/cxx/run-dea.cpp
// In this example, we demonstrate how a Korali experiment can
// be resumed from any point (generation). This is a useful feature
// for continuing jobs after an error, or to fragment big jobs into
// smaller ones that can better fit a supercomputer queue.
//
// First, we run a simple Korali experiment.

#include "korali.hpp"
#include "model/model.h"

int main(int argc, char* argv[])
{
 auto k = Korali::Engine();

 k["Problem"]["Type"] = "Optimization";
 k["Problem"]["Objective"] = "Maximize";

 k["Variables"][0]["Name"] = "X";
 k["Variables"][0]["Lower Bound"] = -10.0;
 k["Variables"][0]["Upper Bound"] = +10.0;

 k["Solver"]["Type"] = "DEA";
 k["Solver"]["Sample Count"] = 10;

 k["General"]["Max Generations"] = 50;
 k["General"]["Console Output"]["Frequency"] = 5;
 k["General"]["Results Output"]["Frequency"] = 5;

 k.setModel(model);
 k.run();

 printf("\n\nRestarting now...\n\n");

 // Now we loadState() to resume the same experiment from generation 5
 k.loadState("_korali_result/s00010.json");

 k.run();

}

File: tutorials/b1-checkpoint-restart/run-dea.py
#!/usr/bin/env python3

## In this example, we demonstrate how a Korali experiment can
## be resumed from any point (generation). This is a useful feature
## for continuing jobs after an error, or to fragment big jobs into
## smaller ones that can better fit a supercomputer queue.

## First, we run a simple Korali experiment.

import sys
sys.path.append('./model')
from model import *

import korali
k = korali.initialize()
e = korali.newExperiment()

e["Problem"]["Type"] = "Evaluation/Direct/Basic"
e["Problem"]["Objective"] = "Maximize"
e["Problem"]["Objective Function"] = model

e["Solver"]["Type"]  = "Optimizer/DEA"
e["Solver"]["Population Size"] = 10
e["Solver"]["Termination Criteria"]["Max Generations"] = 100
e["Solver"]["Termination Criteria"]["Generations Per Run"] = 50

e["Variables"][0]["Name"] = "X"
e["Variables"][0]["Lower Bound"] = -10.0
e["Variables"][0]["Upper Bound"] = +10.0

e["Result Path"] = '_result_run-dea'
e["Console Frequency"] = 5
e["Save Frequency"] = 5
e["Resume Previous"] = True

print('------------------------------------------------------')
print('Now running first 50 generations...')
print('------------------------------------------------------')

k.run(e)

print('------------------------------------------------------')
print('Now running last 50 generations...')
print('------------------------------------------------------')

k.run(e)
